features/steps/steps.go

```diff
@@ -2,63 +2,92 @@ package steps
 	"github.com/ONSdigital/dp-search-data-extractor/models"
 	"github.com/ONSdigital/dp-search-data-extractor/schema"
 	"github.com/ONSdigital/dp-search-data-extractor/service"
 	"github.com/cucumber/godog"
 	"github.com/rdumont/assistdog"
-	"github.com/stretchr/testify/assert"
 )
 
 func (c *Component) RegisterSteps(ctx *godog.ScenarioContext) {
-	ctx.Step(`^these kafka events are consumed:$`, c.theseKafkaEventsAreConsumed)
-	ctx.Step(`^I should receive a kafka response$`, c.iShouldReceiveAKafkaResponse)
+	ctx.Step(`^I send a kafka event to content published topic$`, c.sendKafkafkaEvent)
+	ctx.Step(`^The kafka event is processed$`, c.processKafkaEvent)
+	ctx.Step(`^I should receive the published data$`, c.iShouldReceivePublishedData)
 }
 
-func (c *Component) iShouldReceiveAKafkaResponse() error {
-	content, err := ioutil.ReadFile(outputFilePath)
+func (c *Component) sendKafkafkaEvent(table *godog.Table) error {
+
+	observationEvents, err := c.convertToKafkaEvents(table)
 	if err != nil {
 		return err
 	}
 
-	assert.Equal(c, "Hello, testURL.com", string(content))
+	for _, e := range observationEvents {
+		if err := c.sendToConsumer(e); err != nil {
+			return err
+		}
+	}
 
-	return c.StepError()
+	return nil
 }
 
-func (c *Component) theseKafkaEventsAreConsumed(table *godog.Table) error {
+func (c *Component) processKafkaEvent() error {
+	c.inputData = models.ZebedeeData{
+		DataType: "something",
+		Description: models.Description{
+			CDID:      "123",
+			DatasetID: "456",
+			Edition:   "something",
+		},
+	}
 
-	observationEvents, err := c.convertToKafkaEvents(table)
+	marshelledData, err := json.Marshal(&c.inputData)
 	if err != nil {
-		return err
+		return fmt.Errorf("error marshalling input data")
+	}
+	c.zebedeeClient = &zClientMock.ZebedeeClientMock{
+		CheckerFunc: func(in1 context.Context, in2 *healthcheck.CheckState) error { return nil },
+		GetPublishedDataFunc: func(ctx context.Context, uriString string) ([]byte, error) {
+			return marshelledData, nil
+		},
 	}
-
-	signals := registerInterrupt()
-
 	// run application in separate goroutine
 	go func() {
 		c.svc, err = service.Run(context.Background(), c.serviceList, "", "", "", c.errorChan)
 	}()
 
-	// consume extracted observations
-	for _, e := range observationEvents {
-		if err := c.sendToConsumer(e); err != nil {
-			return err
-		}
-	}
-
-	time.Sleep(300 * time.Millisecond)
+	signals := registerInterrupt()
 
 	// kill application
 	signals <- os.Interrupt
 
+	return err
+}
+
+func (c *Component) iShouldReceivePublishedData() error {
+	outputData := <-c.KafkaProducer.Channels().Output
+
+	var data = &models.SearchDataImport{}
+
+	schema.SearchDataImportEvent.Unmarshal(outputData, data)
+
+	if data.CDID != c.inputData.Description.CDID {
+		return fmt.Errorf("Expected CDID: %v Recevived CDID: %v", c.inputData.Description.CDID, data.CDID)
+	}
+
+	if data.DatasetID != c.inputData.Description.DatasetID {
```

Comment:
Similarly this if statement should be replaced with assert too e.g.
assert.Equal(&c.ErrorFeature, data.DatasetID, c.inputData.Description.DatasetID)

Reply:
Go always suggests not to use external libraries like testify/assert rather just do code there and use the inbuilt code to check for errors. If we look at the various google libraries,auth libraries etc, we would be able to see it .